### PR TITLE
Add Homepage Title + Text

### PR DESCRIPTION
**Pull request by w-gao**
Problem
=======
As a user, 
I want to see the title text and description,
So that I am aware of the tentative event dates.

[Link to Asana Ticket](https://app.asana.com/0/1197738625416321/1198219556358095)

Solution
========
In order for the hero to be two columns and responsive, I structured the hero view with a flexbox. 

- Also updated `.stylelintrc.js` so that we can nest more selectors in SCSS. (Not sure if this is the best way; I did it so the linter won't complain)

Change summary:
---------------
* added `responsive-font` mixin from 2020's website
* updated hero view

Steps to Verify:
----------------
* Check this branch out and start a local server
* Verify that the homepage title and text are visible & responsive

Screenshots:
-----------------------
image

### CODE REVIEW

File: src/views/Hero/Hero.scss [hunk 1]
```diff
@@ -1,11 +1,56 @@
...
+        color: var(--wedgewood__dark);
+      }
+    }
+
+    > p {
+      max-width: 375px;
+      margin: 2em 0;
+
+      @include typography.title-text;
+    }
```
Comment (ilopezro) [inline src/views/Hero/Hero.scss:54]: Lets stray away from doing this. Use classnames instead!

File: src/views/Hero/index.view.tsx [hunk 2]
```diff
@@ -16,11 +16,19 @@ const HeroView: React.FC<HeroProps> = ({ pageName, title, description }) => {
...
+        </div>
+        <div className="HeroView__text">
+          <h1>CruzHacks 2021 /</h1>
+          <h1>
+            <span>{title}</span>
+          </h1>
+          <p>{description}</p>
+          <EmailSubscriptionForm />
+        </div>
+      </div>
```
Comment (ilopezro) [inline src/views/Hero/index.view.tsx:31]: We should give the `h1` and `p` tags classnames. I am also not sure if we want to be nesting `p` tags within `h1` tags; rather, we can nest the `p` tag within a `div` instead. 

```suggestion
        <div className="HeroView__text">
          <h1 className="HeroView__titleText>CruzHacks 2021 /</h1>
          <h1 className="HeroView__titleTextTwo>{title}</h1>
          <p className="HeroView__description">{description}</p>
          <EmailSubscriptionForm />
        </div>
      </div>
```

File: src/stylesheets/_typography.scss [hunk 3]
```diff
@@ -1,14 +1,69 @@
...
+      font-size: $max;
+    }
+  }
+
+  @if $fallback {
+    font-size: $fallback;
+  }
+
+  font-size: $responsive;
 }
```
Comment (ilopezro) [inline src/stylesheets/_typography.scss:48]: Wondering if this would actually go under `_responsiveness.scss` instead of this file. It would make this file much cleaner.

File: src/views/Hero/Hero.scss (hunk 1 above)
Comment (w-gao) [reply src/views/Hero/Hero.scss:54]: Ahh, oops! My bad. 

Although, how would you go about making this nest only once? 

```css
.HeroView {
  &__container {
    // ...
    @include responsiveness.respond(phone) {
      // ...
    }
  }
```

File: src/stylesheets/_typography.scss [hunk 4]
```diff
@@ -1,14 +1,69 @@
+@use "src/stylesheets/responsiveness";
```
Comment (ilopezro) [inline src/stylesheets/_typography.scss:1]: Remember that the root is now the `/src` folder so all the imports will happen from that folder. 

```suggestion
@use "~stylesheets/responsiveness";
```

File: src/views/Hero/Hero.scss (hunk 1 above)
Comment (ilopezro) [reply src/views/Hero/Hero.scss:54]: That's a really good point! How about we increate the nesting to two depth instead of 3?

File: src/stylesheets/_typography.scss [hunk 5]
```diff
@@ -1,14 +1,69 @@
...
+
+  @if $fallback {
+    font-size: $fallback;
+  }
+
+  font-size: $responsive;
 }
 
-@mixin title-info {
+@mixin h1 {
```
Comment (ilopezro) [inline src/stylesheets/_typography.scss:50]: Let's stick with giving mixins more meaningful names. I am not entirely sure what this mixin in used for.

**Review by ilopezro — DISMISSED** [review]
Looks really good overall! Take a look at the comments I left 👍

File: .stylelintrc.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ module.exports = {
             "Classes must be in BEM form like `MyComponent__element--variant`",
         },
       ],
-      "selector-max-compound-selectors": 3,
+      "selector-max-compound-selectors": 1,
```
Comment (ilopezro) [inline .stylelintrc.js]: This should still be 0.

File: src/stylesheets/_typography.scss (hunk 5 above)
Comment (w-gao) [reply src/stylesheets/_typography.scss]: I agree. Is something like `description` good enough? This should be something that can be used in many places.

Comment (ilopezro) [reply src/stylesheets/_typography.scss]: That works for me for now!

File: src/views/Hero/index.view.tsx [hunk 7]
```diff
@@ -16,11 +16,17 @@ const HeroView: React.FC<HeroProps> = ({ pageName, title, description }) => {
...
-      Hello from HeroView
-      {title}
-      {description}
+      <div className="HeroView__container">
+        <div className="HeroView__visual">
+          <PostcardStack pageName={pageName} />
+        </div>
+        <div className="HeroView__text">
+          <h1 className="HeroView__titleText">CruzHacks 2021 /</h1>
+          <h1 className="HeroView__titleTextTwo">{title}</h1>
```
Comment (ilopezro) [inline src/views/Hero/index.view.tsx:25]: Awesome! If we're going with this approach, I'd suggest adding a modifier to that `titleText` class that we have up already. 

```suggestion
          <h1 className="HeroView__titleText--secondary">{title}</h1>
```

In the styling we can nest the following together: 
```scss
&__titleText {
    @include typography.title-text;
    &--secondary {
           color: var(--wedgewood__dark);
    }
  }
```

File: src/views/Hero/index.view.tsx [hunk 8]
```diff
@@ -16,11 +16,17 @@ const HeroView: React.FC<HeroProps> = ({ pageName, title, description }) => {
...
-      <PostcardStack pageName={pageName} />
-      <EmailSubscriptionForm />
-      Hello from HeroView
-      {title}
-      {description}
+      <div className="HeroView__container">
+        <div className="HeroView__visual">
+          <PostcardStack pageName={pageName} />
+        </div>
+        <div className="HeroView__text">
```
Comment (ilopezro) [inline src/views/Hero/index.view.tsx:23]: Super nit-picky, but suggest changing the class name to `HeroView__textContainer` just so we can tell it apart from something else that might be called `text`. This is more of a container.

File: src/views/Hero/index.view.tsx (hunk 7 above)
Comment (w-gao) [reply src/views/Hero/index.view.tsx:25]: Yeah, the properties don't inherit. I think `HeroView__titleText--secondary` looks nicer though so I'll change the class name to that.

File: src/stylesheets/_typography.scss [hunk 9]
```diff
@@ -1,14 +1,26 @@
...
 
-@mixin major-title {
+@mixin title-text {
+  margin: 0;
+  color: var(--white);
   font-family: $font-family-italic;
-  font-size: 50px;
-  color: var(--wedgewood);
+  font-style: italic;
+  font-weight: normal;
```
Comment (ilopezro) [inline src/stylesheets/_typography.scss:11]: If we want normal, italicized text, we can add the fonts for these. I can create a PR and push it against yours if you're not sure how to do this.

Comment (w-gao) [reply src/stylesheets/_typography.scss:11]: Go ahead! I'm not entirely sure how to do this.

Comment (ilopezro) [reply src/stylesheets/_typography.scss:11]: I took a look at the code and we can actually do without those two lines of code. We are calling the italic font with the 500 `font-weight` and that's exactly what the design calls for.

Comment (w-gao) [reply src/stylesheets/_typography.scss:11]: Actually no. `font-style` is necessary for the font to display correctly on iOS (Safari and Chrome) for reasons that I’m not entirely sure. 

My understanding is that `font-style` in `@font-face` specifies the style supported by that font, which is only italics in this case. So we probably have to change the default font style for h1 to match. 

See [this](https://stackoverflow.com/questions/23731492/font-face-does-not-work-properly-for-italic-bold-fonts) questionable Stack overflow answer for reference.

File: src/views/Hero/index.view.tsx [hunk 10]
```diff
@@ -16,11 +16,17 @@ const HeroView: React.FC<HeroProps> = ({ pageName, title, description }) => {
...
-      {title}
-      {description}
+      <div className="HeroView__container">
+        <div className="HeroView__visual">
+          <PostcardStack pageName={pageName} />
+        </div>
+        <div className="HeroView__textContainer">
+          <h1 className="HeroView__titleText">CruzHacks 2021 /</h1>
+          <h1 className="HeroView__titleText--secondary">{title}</h1>
+          <p className="HeroView__description">{description}</p>
```
Comment (tim-nguyen-cs) [inline src/views/Hero/index.view.tsx:26]: Would it also be advisable to change `h1` and `p` to `div` to remove any default styling that we might have to repetitively overwrite?

File: src/views/Hero/Hero.scss [hunk 11]
```diff
@@ -1,11 +1,57 @@
...
+    @include responsiveness.respond(phone) {
+      margin: 1em;
+    }
+  }
+
+  &__visual {
+    // postcard styling
+    max-width: 500px;
+    margin: 0 auto;
+  }
```
Comment (tim-nguyen-cs) [inline src/views/Hero/Hero.scss:36]: Is this class also necessary when the `PostcardStack` comprises its own component?

Comment (ilopezro) [reply src/views/Hero/Hero.scss:36]: This might be necessary because remember that `PostcardStack` will only style the component itself independent of where it goes on the hero page.

File: src/stylesheets/_typography.scss (hunk 9 above)
Comment (ilopezro) [reply src/stylesheets/_typography.scss:11]: To piggy back [Tim's comment](https://github.com/CruzHacks/cruzhacks-2021-website/pull/9#discussion_r507198544), I'm wondering if this might be due to some underlying default styling by the `h1` and `p` tags.

File: src/views/Hero/Hero.scss [hunk 12]
```diff
@@ -1,11 +1,57 @@
+@use "~stylesheets/typography";
+@use "~stylesheets/responsiveness";
+
 .HeroView {
-  height: 100%;
+  display: flex;
+  overflow: hidden;
+  justify-content: center;
+  min-height: 100%;
```
Comment (tim-nguyen-cs) [inline src/views/Hero/Hero.scss:8]: When I inspect the code this style is actually overriden by `min-height: 90%` - do we still need to include this?

File: src/views/Hero/Hero.scss [hunk 13]
```diff
@@ -1,11 +1,57 @@
...
-    height: 90%;
+    min-height: 90%;
     border-bottom-left-radius: 50px;
     border-bottom-right-radius: 50px;
     box-shadow: 0 10px 16px rgba(0, 0, 0, 0.25);
   }
+
+  &__container {
+    display: flex;
+    flex-direction: row;
```
Comment (tim-nguyen-cs) [inline src/views/Hero/Hero.scss:20]: `flex-direction: row` is declared by default, so I'm not sure if we need this.

File: src/views/Hero/Hero.scss [hunk 14]
```diff
@@ -1,11 +1,57 @@
...
+    flex-wrap: wrap;
+    justify-content: center;
```
Comment (tim-nguyen-cs) [inline src/views/Hero/Hero.scss:22]: It looks like each child element within this container is being sized to-scale (i.e. the widths and margins of the two child elements fill up the entire container). In this case, would we still need this property?

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (w-gao) [reply src/views/Hero/index.view.tsx:26]: Perhaps. Would it make the TSX file less readable if we don't use the heading and `p` tags for text? To me `div` tags are more for containers but that's a personal preference.

File: src/stylesheets/_typography.scss (hunk 9 above)
Comment (w-gao) [reply src/stylesheets/_typography.scss:11]: That doesn't seem to be the case. The font is still not showing up correctly on mobile with the `div` tag.

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: We could also take advantage of the `h1, p, ...` elements by redeclaring their default styles in one of the base stylesheets!

Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: Another route would be to completely make text elements generic with `span`, although that's a pretty big compromise to HTML readability

File: src/stylesheets/_typography.scss (hunk 9 above)
Comment (mocklee) [reply src/stylesheets/_typography.scss:11]: if the font isn't being rendered at all, it probably points to an issue with the `font-family` reference. 

what's the relevance of appending `"sans-serif"` to the variable declaration here?

```
$font-family-base: "Montserrat", "sans-serif";
$font-family-italic: "Montserrat Italic", "sans-serif";
```

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (tim-nguyen-cs) [reply src/views/Hero/index.view.tsx:26]: I think that's a great recommendation - we can forgo class names in exchange for general header formatting.

File: src/views/Hero/Hero.scss (hunk 11 above)
Comment (tim-nguyen-cs) [reply src/views/Hero/Hero.scss:36]: Ahh, that makes sense - we'll have to re-ensure that the `PostcardStack` composes only of itself rather than the container surounding it.

File: src/stylesheets/_typography.scss (hunk 9 above)
Comment (ilopezro) [reply src/stylesheets/_typography.scss:11]: I added those in, but we can take those out. It was the name of the font on the site that I used to download the fonts.

Comment (w-gao) [reply src/stylesheets/_typography.scss:11]: I might've worded it weird. The font is being rendered, but it's not Montserrat italic on mobile. It works on desktop though. 

Also, I think `"sans-serif"` should be without quotes?

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (ilopezro) [reply src/views/Hero/index.view.tsx:26]: @tim-nguyen-cs we want to stay organized with class names. We don't want to eliminate classmates. What we can do is add a general mixin for an `h1`, `p`, etc tags and include that mixin for each div.

Comment (w-gao) [reply src/views/Hero/index.view.tsx:26]: @tim-nguyen-cs I think @ilopezro wants to use class names though (https://github.com/CruzHacks/cruzhacks-2021-website/pull/9#discussion_r506986915), unless I'm misunderstanding your suggestion?

File: src/stylesheets/_typography.scss (hunk 9 above)
Comment (mocklee) [reply src/stylesheets/_typography.scss:11]: yep, `sans-serif` is a generic/fallback font-family, which is a sort of baseline that CSS sees as an actual property (like colors) 
https://developer.mozilla.org/en-US/docs/Web/CSS/font-family

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: @ilopezro I don't think we need to introduce unnecessary abstraction if we can avoid it— mixins are useful, but if we can select the HTML elements themselves why bother? I do agree that CSS classes are important & useful for granular styling, but if we can establish baseline style formatting let's do it.

Comment (ilopezro) [reply src/views/Hero/index.view.tsx:26]: Especially in the `Hero` component, we don't really need the usage of mixins that much. We will be reusing hero component throughout our views. Let's stick with class names + defining everything at the Hero level since I don't see us reusing those outside of the hero component in order to avoid over abstraction! how does that sound y'all? I do still think we should change the `h1` tags into `div` tags and style the text under a `span` tag (or simply under the `div` tag), but that's just me.

Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: @ilopezro divs containing spans would get the job done, but it blows up the HTML file size if we introduce that redundancy. Plus it seems that using traditional HTML element references promotes readability (let's keep this source code interesting so it can be workshop material)

File: src/views/Hero/Hero.scss (hunk 12 above)
Comment (w-gao) [reply src/views/Hero/Hero.scss:8]: We probably don't want to remove it here since it's a fallback height. If all Hero components have the same height then we can just set it here once and not override it. 

@ilopezro Thoughts?

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (w-gao) [reply src/views/Hero/index.view.tsx:26]: Have we reached a consensus here?

File: src/views/Hero/Hero.scss (hunk 12 above)
Comment (ilopezro) [reply src/views/Hero/Hero.scss:8]: Correct that's the intended purpose of having the different class names within the hero page as well to prepare for the future views.

File: src/views/Hero/index.view.tsx (hunk 10 above)
Comment (tim-nguyen-cs) [reply src/views/Hero/index.view.tsx:26]: Just putting in my two cents, I am recommending these two approaches:

1. `div` tags with class names (alternatively also `span` tags for inline styling)
2. `h*` and `p` tags without class names (because the overlap would be redundant/superfluous)

This is just based on the development experience I've been encompassing, but let me know what the "law of the land" says.

Comment (ilopezro) [reply src/views/Hero/index.view.tsx:26]: I like the idea of using `h*` and `p` tags without class names but the issue with that is that the CSS will get hard to follow and readability will be low. Also, by adding defaulting values for these tags as well, we are making a big assumption that all those tags will be rendered the same and I don't like that approach. I'd rather just define the `div` to what I want it to look like and using `span` or just plain text within the div itself.

Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: why not both & have classnames + simple elements? we can reset the default styles for established HTML elements to our own design pattern using base stylesheets & use granular classnames as well since CSS cascades

Comment (ilopezro) [reply src/views/Hero/index.view.tsx:26]: I think that would defeat the purpose of having CSS files within each component and view if we're defining basic elements at the base level. I think let's stick with class names since this is the only place where I foresee we're using these specific styles. These are hero specific so I don't see why we should redefine these styles higher up in the tree. Let's keep these styles component specific w/ class names. 

Again, my suggestion here would be `div` and a `p` nested within the divs but `h1` also works for me. I usually don't use the header tags and nest a `span` or `p` within a `div`.

Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: here's a demo of how cascading base HTML elements into `div` containers can help us cut some abstraction: https://codepen.io/mocklee/pen/wvWzxzV

Comment (ilopezro) [reply src/views/Hero/index.view.tsx:26]: Ok i see that. I personally don't like this approach because to reiterate, we'd be making a huge assumption about our tags if we move it further up the tree. The further down you to you have to set and unset props and that seem unnecessary. If there is code that we can refactor, we can should create mixins instead. We should make full use of the SCSS magic.

Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: updated the demo to demonstrate nested classnames. I'd say that refactors w/ adding granular classnames later on are tolerable, as long as there aren't too many distinct changes. 

also, it seems that mixins are being siloed to serve more as functions— [the Sass standard also has @extend available, which is syntactical sugar for short mixin declaration](https://sass-lang.com/documentation/at-rules/extend).

Comment (ilopezro) [reply src/views/Hero/index.view.tsx:26]: Yeah I noticed the updated Codepen environment you've got up. If we go with pure classnames, there is no need for that mixin to unset the properties. That becomes redundant when you redefine them inside the another component. I wouldn't want to get too much technical debt if we over refactor and then later on find out we're doubling up on code because we've over refactored at the beginning. Let's stick with classnames for now. When there is a place to refactor, lets go for it!

Comment (mocklee) [reply src/views/Hero/index.view.tsx:26]: [SCSS psuedoclasses](https://css-tricks.com/the-sass-ampersand/) only run one level deep, don't overthink nesting! In general, CSS inheritance isn't hard; but nuanced— it depends on both local pseudoclasses & general DOM (HTML) tree structure. if classnames make it easier, we'll go with that.

if `%moreStyles` seems redundant, please demo an improvement!

**Review by ilopezro — APPROVED** [review]
This looks really good overall man! Great work 🎉
